lib/client.js

```diff
@@ -375,18 +370,7 @@ class Client extends EventEmitter {
       })
     }
 
-    // TODO wait for queued requests as well, not just in-flight?
-
-    // TODO test this
-    if (
-      this[kQueue].length() === 0 &&
-      this[kCallbacks].length === 0 &&
-      this.socket
-    ) {
-      this.socket.end()
-      // TODO wait util socket is closed before setting to null?
-      this.socket = null
-    }
```

Comment:
Don't end the socket if _lastBody exists. Could use allowHalfOpen but there is not much advantage to that, just destroy the socket once we are done.